### ISSUE TITLE
1.9b1 app download link in here

### ISSUE THREAD

**Issue by gingerbeardman** [Ascoware/get-iplayer-automator#1]
Hi,

Here's @skovatch 's build of 1.9b1 hosted on dropbox: [Get iPlayer Automator.zip](https://www.dropbox.com/s/ozouh99l5q2y5ux/Get%20iPlayer%20Automator.app.zip?dl=1)

Important: delete this folder if you're having problems
`~/Library/Application Support/Get iPlayer Automator/`

Enjoy and thanks again to @skovatch

**Comment by botchla** [comment]
The link was there, but the author of the other releases (dinkypumpkin) took it down because he said that without the source code it violated the terms of the open source license...

**Comment by gingerbeardman** [comment]
@botchla just put a link to a build I have just done, see above.

I guess dunkypumpkin didn't look for the source code diligently enough. I found it quite easily!

**Comment by moshimoshiii** [comment]
Thanks so much for this new build!

But unfortunately it doesn't seem to be working for me. Here's the contents of the log, hope it's helpful. I'm on a MBP running 10.8.5.

```
BBC Download (ID=b08qdxk7): Downloading Mexico: Earth's Festival of Life: Series 1 - Mountain Worlds
WARNING: Please download and run latest installer or install the JSON::PP Perl module for more accurate programme metadata.
WARNING: Could not determine PID type. Trying to record PID directly.
INFO: Trying pid: b08qdxk7 using type: tv
INFO: Trying to download PID using type tv
INFO: pid not found in tv cache
INFO: 1 Matching Programmes
WARNING: Please download and run latest installer or install the JSON::PP Perl module for more accurate programme metadata.
WARNING: No programmes are available for this PID with version(s): default (available versions: original)
BBC Download (ID=b08qdxk7): Mexico: Earth's Festival of Life: Series 1 - Mountain Worlds Failed
```

**Comment by Themanwithoutaplan** [comment]
@moshimoshiii looks like you need to update your system's Perl with either MacPorts or Homebrew.

**Comment by gingerbeardman** [comment]
@moshimoshiii I've taken the liberty of starting a new issue about Perl, as it would be good for the app to automatically detect out-of-date versions and prompt users to update and explain how. Please continue your discussion there and we can help get you running. 👍

**Comment by chazf1** [comment]
The new GIA 1.9b1 doesn't appear to index ITV programmes, but downloads via webpage only.
Is this correct?
I note that the previous version GIA 1.8.8-pre5 does still indexes current ITV content.
Any chance of resisting this or should we just plan to download via webpage in the future?

**Comment by gingerbeardman** [comment]
@chazf1 i can't say, but please start a new issue about this, cheers

**Comment by gingerbeardman** [comment]
@chazf1 since 1.8.7

> If ITV cache updating dies again as it did for 6+ months last year, you now have access to an alternate ITV indexing implementation (from @LFS1). Enable it in the Advanced tab in preferences.

Try that?

**Comment by Duffledorf** [comment]
A big THANK YOU to skovatch for the updated version. working great here! One minor question, previously GiA created files and folders with spaces rather than underlines is there a way to revert to that?

ie:  'Radio 2 Playlists' rather than 'Radio_2_Playlists'

thanks again all

**Comment by Themanwithoutaplan** [comment]
@Duffledorf you need to set the preference to use Kodi/XBMC style filenames, ie. preferences from the previous version weren't persisted.

**Comment by Roalnderson** [comment]
Seems to be working great with me!!! Many thanks. I'm guessing there will be some regular updates over the next few days and so will it update automatically, or should i 'check for updates' via the app, or just check in here regularly? Thanks again to everyone involved, @skovatch @gingerbeardman @dinkypumpkin

**Comment by vertigo4** [comment]
I'm also having problems with the new build. Error message: 'None of the download formats you specified were available for this show etc', even though I have set GIA to use all available formats.

Here's the log:

Get iPlayer Automator 1.9b1 Initialized.

AppController: Updating Programme Index Feeds...

AppController: Index was Up-To-Date.

INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used

AppController: Starting Downloads

Downloading Show 1/1:

BBC Download (ID=b08q8jcy): Downloading Babs - Babs
INFO: Episode PID detected
INFO: Trying pid: b08q8jcy using type: tv
INFO: Trying to download PID using type tv
INFO: pid not found in tv cache
INFO: 1 Matching Programmes
WARNING: Some recording modes were remapped to new values
WARNING: Input mode list remapped from 'dafhigh,dafstd,hlsvhigh,hlshd,hlsstd,hlslow,' to 'dafhigh,dafstd,hlsvhigh,hlshd,hlsstd,hlsxstd,'
WARNING: Please update your preferences
INFO: Checking existence of original version
INFO: No specified modes (dafhigh,dafstd,hlsvhigh,hlshd,hlsstd,hlsxstd) available for this programme with version 'original'
INFO: Try using --modes=dvfhigh,dvflow,dvfhd,dvfxhigh,dvfxsd,dvfsd,hvfhigh,hvflow,hvfxsd,hvfsd,hvfxhigh,hvfhd
ERROR: Failed to record 'Babs - - (b08q8jcy)'
REASON FOR FAILURE: None of the modes in your download format list are available for this self.show.
Try adding more modes.
BBC Download (ID=b08q8jcy): Babs - Babs Failed

AppController: Downloads Finished

**Comment by OldTrout** [comment]
Thanks for all the hard work guys. Despite happy mail above it does not work for me. I'm on OSX 10.12.4. App loads, fills up cache and can find e.g. Hinterland (BBC4), and can add this to queue. It can also get Hinterland from webpage. However when a download of the queue is started it fails immediately (with notification). The progress bar at the bottom is filled with blue with Time until Start showing as 25 or so minutes.
Failed Download Solution pane gives diagnosis as "Problem unknown. Please submit a bug report from the application menu." I can't find any bug report in the menus and the Window/Log is greyed out so I can't look there.
How can I get anything useful to you?

**Comment by ian02054** [comment]
Me too.. log says unable to find file segment.

**Comment by OldTrout** [comment]
Duh. Missed log button on main window.
Log referred to ~/Library/Application Support/Get iPlayer Automator/ so renamed this Old, quit the app and restarted. Now working!
It doesn't find ITV stuff but it can use current webpage for ITV. Odd?

**Comment by gingerbeardman** [comment]
@OldTrout the ITV issue has been reported above by somebody else. Let's wait for @skovatch to wake up and see what he says.

**Comment by Rock-Hunter** [comment]
BIG THANK YOU to @gingerbeardman & @skovatch for making this work.

**Comment by gingerbeardman** [comment]
@Rock-Hunter the effort was all @skovatch I am just a humble user like yourselves who decided to right some wrongs 🤡

**Comment by botchla** [comment]
Having some success with this, and I'm very grateful to all involved. There are a couple of things I've noticed and I don't know whether it's just me:

1) The PVR/series link items are being picked up but it's adding many previously downloaded shows to the queue and then failing with the error that they're in the history.
2) Some downloads are failing because the quality isn't found or there's a missing segment. This appears to be a get_iplayer issue (although, actually, it's a BBC stream issue) and isn't because of a problem with GiA.

**Comment by vertigo4** [comment]
Following my post above I deleted GIA, reinstalled and changed to UK proxy/ IP address and now all seems to work very well. 

Many, many, many thanks to those who have done the hard work on this great app.

**Comment by Lordsnooty05** [comment]
I'm in Portugal on holiday at the moment....so looking forward to using this when I get home next week but just wanted to say a word of thanks and support to Skovatch for all the work they have done in bringing back GIA

**Comment by Duffledorf** [comment]
> @Duffledorf you need to set the preference to use Kodi/XBMC style filenames, ie. preferences from the previous version weren't persisted.

 @ Themanwithoutaplan - yes had done this - and restarted, but doesn't seem to make a difference.

minor problem - it can wait.

cheers

**Comment by Themanwithoutaplan** [comment]
@Duffledorf worked fine for me without even a restart and even while a queue was running. Would be nice to have this as the default, though. Along with subtitles.

**Comment by pmhird** [comment]
Thank you very much to all concerned who restored functionality to this excellent app. Just one snag for me, not a complaint but if you have any advice Ill gratefully accept. The PVR doesn't pick up new shows for me and entering a show in the search box seems to cause the app to update series links and then returns no shows. If i use the "Use current webpage" button it works fine however. 

In terms of install, I erased my old GIA from my applications folder, and I moved the new one into applications and ran it from there. I'm on OSX 10.9.5. 

Any ideas how to resort PVR and search functionality? Do I need to do a clean install and if so can you give me a guide? Many many thanks once again.

**Comment by dvioletta** [comment]
Thank you very much for taking the time to get this sorted out and put back together.

I have encountered an issue that the series link when I try to refresh it constantly reshows programs it has already downloaded.

Example first run through - Added Doctor Who: Series 10 - Thin Ice from PVR queue.
Downloaded and added to iTunes successfully.
Force Refreshed Cashe but the original list still remains.

**Comment by pmhird** [comment]
dvioletta- hasn't it always done that? The completed list will disappear next time you download a new show i think. If you try to add a new show the completed ones will clear from the list I'm pretty sure...

**Comment by dvioletta** [comment]
It has never behaved like this for me before, it used to clear the screen when the cache refreshed and never added back in already downloaded shows as it is at present.

**Comment by ian02054** [comment]
there is an option in preferences under the advanced tab to not show already downloaded files. this may be on by default.

**Comment by botchla** [comment]
The issue @dvioletta is describing is what I mentioned earlier and isn't affected by the Advanced tab preference that @ian02054 just mentioned. If I'm right, the issue is that each time the PVR runs, it populates the queue with all the right shows, but it doesn't ignore previously downloaded shows from the history.

**Comment by kiwibob** [comment]
Thank you for this. The app worked straight out of the box for me. Hinterland binge watching for me this evening!

**Comment by skovatch** [comment]
Version 1.9b2 is up with ITV searching restored. It should also fix problems with already-downloaded shows, as I fixed a problem with the application support path.

**Comment by A-viewer** [comment]
Thanks for all the hard work.

**Comment by mh57** [comment]
Many thanks for resurrecting the software. FYI shows download, but often in multiple formats and frequently fail to convert to mp4 format.

**Comment by OldTrout** [comment]
I'm a new boy on here but very appreciative of all the effort that goes into getting GiA operating fully. Thank you.
@skovatch, when you say "Version 1.9b2 is up", up where?

**Comment by Rock-Hunter** [comment]
I'm having a slight problem whereby certain Radio 4 programmes are coming up "None of the download formats you specified were available for this show" despite my having selected all available options in settings. Any advice would be gratefully appreciated.

**Comment by Rock-Hunter** [comment]
@OldTrout click on the download link at the top of the page.

**Comment by OldTrout** [comment]
Thanks @Rock-Hunter but I tried that. A 2 has appeared in the title but the version number of the app is still 1.9b1 (and ITV searching is not restored).

OT @kiwibob I binged on Hinterland series 3 last week. The national coverage began as you know last Saturday. However last night I started again with series 1. It is very good and the links to the later episodes are all there.

**Comment by nigelbb** [comment]
Thank you very much for updating this wonderful program so that it works again. My uses are very limited in that 99% of my usage is downloading programmes from Radio 4 & putting them in iTunes so I can listen to them on my iPhone when I am in the car. During the days that GIA wasn't working I was pleasantly surprised to discover that using the command line get_iplayer wasn't that awful but the GUI automated version does make things that bit easier. Once again thanks.

**Comment by nigelbb** [comment]
It's a minor point but I've just realised that all the files now have underscores instead of spaces in the filenames so with the older version I would download
In Our Time.s01e761 The Battle of Lincoln 1217.m4a
with the new version the filename is
In_Our_Time.s01e761.The_Battle_of_Lincoln_1217.m4a
The containing folders also now have underscores e.g.
In_Our_Time not In Our Time

Is there any way of restoring the original behaviour so the new files match all my previously downloaded shows?

**Comment by Duffledorf** [comment]
@OldTrout - You need to download from this page

**Comment by OldTrout** [comment]
@Duffledorf thanks but as I put four posts up the link at the top of this page is still version 1.9b1.

**Comment by botchla** [comment]
@OldTrout Sorry to repeat @Duffledorf's post, but as he said, the latest release is here: https://github.com/Ascoware/get-iplayer-automator/releases and here's a direct link: https://github.com/Ascoware/get-iplayer-automator/releases/download/1.9b2/Get.iPlayer.Automator.app.zip

**Comment by Duffledorf** [comment]
I have updated to 1.9b2, but am still having the issue re the underscores that I mentioned above and that is detailed in @nigelbb post above. I have run the same show through twice with the "Use Kodi (XBMC) Naming:" option off and then on; I have copies of the logs of both if they are of any use to anyone who may be able to identify the issue. The best that I can identify is that "((dafstd))" appears a few times in the log when the naming option is off; and is missing when it is on.

@skovatch are the logs any use to you?

**Comment by OldTrout** [comment]
@botchla and @Duffledorf thank you. I didn't think I was being thick. I am using Safari 10.1 and the link in @Duffledorf's post does not show! If I use Firefox 50.0 or Safari Technology Preview it does. 
I have Ghostery, uBlock, InvisibleHand, Incognito and ClickToPlugin running in Safari but nothing in safari Technology Preview.

**Comment by neilstuart** [comment]
Thank you Works Great! OS 10.12.5

**Comment by jezzastern** [comment]
I am still on Snoe Leopard. I had the previous version which was updated using terminal according to the update instructions for Snow Leopard. Any news on whether the new incarnation of GIA will support Snow Leopard?

**Comment by prvincent** [comment]
Many, many thanks to @skovatch for the hard work put into restoring the functionality of GIA. It's an enormous relief to be able to download and keep programmes once again. My IT skills are far too outdated to offer any meaningful assistance here, but if there's ever an appeal for financial donations, I would certainly chip in.

**Comment by broken42** [comment]
On 10.12.4 with 1.9b2 I'm getting inconsistent results. I have successfully downloaded latest Doctor Who for example but am unable to do so with any of the Pointless episodes I've tried.  Download starts, runs for a bit, then aborts. Log has a "File segment not available from server [xx]" message. Log file is attached. Never had this issue with 1.8.7 or earlier.

I reiterate all the thank you's to developers past and present for this, and will gladly contribute funds to keep this going.
[GiA log.pdf](https://github.com/Ascoware/get-iplayer-automator/files/988201/GiA.log.pdf)
